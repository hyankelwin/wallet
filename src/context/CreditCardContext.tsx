import React, { createContext, useState, useCallback, useContext } from 'react';
import {
  CreditCard,
  SideProps,
  FlagCard,
} from '../common/interfaces/CreditCard.interface';
import { getTimeOut } from '../utils/getTimeOutSimulateApi';

interface CreateCreditContext {
  interfacefValuesCard: CreditCard;
  side: SideProps;
  fieldFocused: string;
  imageSelected: string;
  flagCard: FlagCard;
  isLoading: boolean;
  error: boolean;
  creditCard: CreditCard;
  handleClearCard(): void;
  handleSetValuesCreditCard(field: string, value: string): void;
  handleTransitionCreditCard(name: string, value: boolean): void;
  handleFocusField(field: string): void;
  handleSaveImageCard(image: string): void;
  handleSaveFlagCard(value: FlagCard): void;
  handleCreateCreditCard(values: CreditCard): Promise<void>;
}

const CreditCardContext = createContext<CreateCreditContext>(
  {} as CreateCreditContext,
);

const CreditCardProvider: React.FC = ({ children }) => {
  const [interfacefValuesCard, setInterfacefValuesCard] = useState<CreditCard>(
    {} as CreditCard,
  );
  const [side, setSide] = useState<SideProps>({} as SideProps);
  const [fieldFocused, setFieldFocused] = useState('');
  const [imageSelected, setImageSelected] = useState('');
  const [flagCard, setFlagCard] = useState<FlagCard>({} as FlagCard);
  const [isLoading, setLoading] = useState<boolean>(false);
  const [error, setError] = useState<boolean>(false);
  const [creditCard, setCreditCard] = useState<CreditCard>({} as CreditCard);

  const handleSetValuesCreditCard = useCallback(
    (field: string, value: string) => {
      setInterfacefValuesCard({ ...interfacefValuesCard, [field]: value });
    },
    [interfacefValuesCard, setInterfacefValuesCard],
  );

  const handleTransitionCreditCard = useCallback(
    (name: string, value: boolean) => {
      setSide({ name, value });
    },
    [],
  );

  const handleFocusField = useCallback((field: string) => {
    setFieldFocused(field);
  }, []);

  const handleSaveImageCard = useCallback((image: string) => {
    setImageSelected(image);
  }, []);

  const handleSaveFlagCard = useCallback((flag: FlagCard) => {
    setFlagCard(flag);
  }, []);

  const handleCreateCreditCard = useCallback(async (values: CreditCard) => {
    setLoading(true);
    setError(false);

    const response = values;
    await setCreditCard(response);

    await getTimeOut(10);
    setLoading(false);
  }, []);

  const handleClearCard = useCallback(() => {
    setCreditCard({} as CreditCard);
    setInterfacefValuesCard({} as CreditCard);
    setFlagCard({} as FlagCard);
    setImageSelected('');
  }, []);

  return (
    <CreditCardContext.Provider
      value={{
        creditCard,
        interfacefValuesCard,
        side,
        fieldFocused,
        imageSelected,
        flagCard,
        isLoading,
        error,
        handleClearCard,
        handleSaveFlagCard,
        handleFocusField,
        handleSaveImageCard,
        handleSetValuesCreditCard,
        handleTransitionCreditCard,
        handleCreateCreditCard,
      }}
    >
      {children}
    </CreditCardContext.Provider>
  );
};

function useCreditCard(): CreateCreditContext {
  const context = useContext(CreditCardContext);

  return context;
}

export { CreditCardProvider, useCreditCard };
