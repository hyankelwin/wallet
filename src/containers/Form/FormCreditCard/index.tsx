import React, { useRef, useCallback } from 'react';
import { Form } from '@unform/web';
import { FiUser, FiKey } from 'react-icons/fi';
import { FormHandles } from '@unform/core';
import * as Yup from 'yup';
import {
  ContentForm,
  FooterForm,
  Column,
  ContentImage,
  Image,
} from '../styles';
import Input from '../../../components/Input';
import InputSelect from '../../../components/InputSelect';
import Button from '../../../components/Button';
import getValidationErrors from '../../../utils/getValidationErrors';
import Color from '../../../styles/color';
import { useCreditCard } from '../../../context/CreditCardContext';
import { months, years } from '../../../utils/mountExpirationsDate';
import CardsImages from '../../../constants/images.constant';
import { useForm } from '../../../hooks/useForm';

const FormCreditCard: React.FC = () => {
  const { isFlag, handleTypeFlagCard } = useForm();

  const {
    handleSetValuesCreditCard,
    handleTransitionCreditCard,
    handleFocusField,
    handleCreateCreditCard,
    handleSaveFlagCard,
    handleSaveImageCard,
    imageSelected,
  } = useCreditCard();

  const formRef = useRef<FormHandles>(null);

  const handleSubmit = useCallback(
    async values => {
      const handleValidateMonth = (year: string, month: string) => {
        const monthNow = new Date().getMonth() + 1;
        const yearNow = new Date().getFullYear();

        if (year === String(yearNow)) {
          return !(Number(month) < monthNow);
        }

        return true;
      };

      try {
        formRef.current?.setErrors({});
        const schema = Yup.object().shape({
          number: Yup.string()
            .required('Card number is required')
            .test('$isFlag', 'Enter a valid number', () => isFlag),
          name: Yup.string()
            .required('Card number is required')
            .min(6, 'Minimum 6 digits'),
          cvv: Yup.string()
            .required('CVV is required')
            .max(4, 'Maximum 6 digits'),
          month: Yup.string()
            .required('Month is required')
            .test(
              '$isMonthValidated',
              'Month selected is less that month current',
              () => handleValidateMonth(values.year, values.month),
            ),
          year: Yup.string().required('Year is required'),
        });
        await schema.validate(values, {
          abortEarly: false,
        });

        await handleCreateCreditCard(values);
      } catch (error) {
        const errors = getValidationErrors(error as Yup.ValidationError);
        formRef.current?.setErrors(errors);
      }
    },
    [isFlag, handleCreateCreditCard],
  );

  return (
    <Form ref={formRef} onSubmit={handleSubmit}>
      <ContentImage>
        {CardsImages.map(item => (
          <Image
            data-testid="image-card-test"
            key={item}
            onClick={() => {
              handleSaveImageCard(item);
            }}
            img={item}
            imageSelected={imageSelected}
          />
        ))}
      </ContentImage>
      <ContentForm>
        <Column>
          <span>Card number</span>
          <Input
            mask="credit-card"
            name="number"
            icon={FiKey}
            data-testid="input-number"
            onFocus={() => handleFocusField('number')}
            onChange={async data => {
              const type = await handleTypeFlagCard(data.target.value);
              await handleSaveFlagCard(type);
              await handleSetValuesCreditCard('number', data.target.value);
            }}
          />
        </Column>
        <Column>
          <span>Card name</span>
          <Input
            mask="only-string"
            name="name"
            data-testid="input-name"
            icon={FiUser}
            style={{ textTransform: 'uppercase' }}
            onChange={data =>
              handleSetValuesCreditCard('name', data.target.value)
            }
            onFocus={() => handleFocusField('name')}
          />
        </Column>
        <Column>
          <span>Expiration date</span>
          <div className="row">
            <div className="expiration-input">
              <InputSelect
                items={months()}
                data-testid="input-month"
                title="Month"
                small
                name="month"
                onChange={data => {
                  handleSetValuesCreditCard(
                    'expiration_month',
                    data.target.value,
                  );
                }}
              />
              <InputSelect
                items={years()}
                title="Year"
                data-testid="input-year"
                name="year"
                small
                onChange={data =>
                  handleSetValuesCreditCard(
                    'expiration_year',
                    data.target.value,
                  )
                }
              />
            </div>
            <div className="cvv">
              <Input
                onFocus={() => handleTransitionCreditCard('back', true)}
                onBlur={() => handleTransitionCreditCard('front', false)}
                onChange={data =>
                  handleSetValuesCreditCard('cvv', data.target.value)
                }
                placeholder="CVV"
                small
                data-testid="input-cvv"
                name="cvv"
              />
            </div>
          </div>
        </Column>
        <FooterForm>
          <Button
            type="submit"
            color={Color.white}
            background={Color.primary.default}
          >
            Create new card
          </Button>
        </FooterForm>
      </ContentForm>
    </Form>
  );
};

export default FormCreditCard;
