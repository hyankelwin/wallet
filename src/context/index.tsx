import React from 'react';

import { CreditCardProvider } from './CreditCardContext';

const AppProvider: React.FC = ({ children }) => (
  <CreditCardProvider>{children}</CreditCardProvider>
);

export default AppProvider;
