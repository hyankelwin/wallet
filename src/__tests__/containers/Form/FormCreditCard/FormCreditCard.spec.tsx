import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import FormCreditCard from '../../../../containers/Form/FormCreditCard/index';
import { mockCreditCard } from '../../../../__mocks__/creditCard.mock';
import { useCreditCard } from '../../../../context/CreditCardContext';

const mockedError = jest.fn();
const mockedHistoryPush = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useHistory: () => ({
      push: mockedHistoryPush,
    }),
    Link: ({ children }: { children: React.ReactNode }) => children,
  };
});

jest.mock('../../../../hooks/useForm', () => {
  return {
    useForm: () => ({
      handleTypeFlagCard: jest.fn(),
      handleImageSelected: jest.fn(),
      isFlag: true,
      imgCard: '10.jpeg',
    }),
  };
});

jest.mock('../../../../context/CreditCardContext');

describe('FormCreditCard container', () => {
  it('should be able to add new credit card on FormCreditCard container with success', async () => {
    (useCreditCard as unknown as jest.Mock).mockReturnValue({
      handleSetValuesCreditCard: () => jest.fn(),
      handleTransitionCreditCard: () => jest.fn(),
      handleFocusField: () => jest.fn(),
      handleSaveFlagCard: () => jest.fn(),
      handleCreateCreditCard: () => jest.fn(),
      creditCard: mockCreditCard,
    });

    const { getByTestId, getByText, asFragment } = render(<FormCreditCard />);

    const numberField = getByTestId('input-number');
    const nameField = getByTestId('input-name');
    const monthField = getByTestId('input-month');
    const yearField = getByTestId('input-year');
    const cvvField = getByTestId('input-cvv');
    const buttonElement = getByText('Create new card');

    fireEvent.focus(numberField);
    fireEvent.focus(nameField);
    fireEvent.focus(cvvField);
    fireEvent.blur(cvvField);

    fireEvent.change(numberField, { target: { value: mockCreditCard.number } });
    fireEvent.change(nameField, { target: { value: mockCreditCard.name } });
    fireEvent.change(monthField, {
      target: { value: mockCreditCard.expiration.month },
    });
    fireEvent.change(yearField, {
      target: { value: mockCreditCard.expiration.year },
    });
    fireEvent.change(cvvField, { target: { value: mockCreditCard.cvv } });

    fireEvent.click(buttonElement);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should be able to add new credit card on FormCreditCard when year equal current', async () => {
    (useCreditCard as unknown as jest.Mock).mockReturnValue({
      handleSetValuesCreditCard: () => jest.fn(),
      handleTransitionCreditCard: () => jest.fn(),
      handleFocusField: () => jest.fn(),
      handleSaveFlagCard: () => jest.fn(),
      handleCreateCreditCard: () => jest.fn(),
      creditCard: mockCreditCard,
    });

    const { getByTestId, getByText, asFragment } = render(<FormCreditCard />);

    const numberField = getByTestId('input-number');
    const nameField = getByTestId('input-name');
    const monthField = getByTestId('input-month');
    const yearField = getByTestId('input-year');
    const cvvField = getByTestId('input-cvv');
    const buttonElement = getByText('Create new card');

    fireEvent.focus(numberField);
    fireEvent.focus(nameField);
    fireEvent.focus(cvvField);
    fireEvent.blur(cvvField);

    fireEvent.change(numberField, { target: { value: mockCreditCard.number } });
    fireEvent.change(nameField, { target: { value: mockCreditCard.name } });
    fireEvent.change(monthField, {
      target: { value: mockCreditCard.expiration.month },
    });
    fireEvent.change(yearField, {
      target: { value: new Date().getFullYear() },
    });
    fireEvent.change(cvvField, { target: { value: mockCreditCard.cvv } });

    fireEvent.click(buttonElement);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should be able to simulate click image card with success', async () => {
    (useCreditCard as unknown as jest.Mock).mockReturnValue({
      handleSetValuesCreditCard: () => jest.fn(),
      handleTransitionCreditCard: () => jest.fn(),
      handleFocusField: () => jest.fn(),
      handleSaveFlagCard: () => jest.fn(),
      handleSaveImageCard: () => jest.fn(),
      handleCreateCreditCard: () => jest.fn(),
      creditCard: mockCreditCard,
      imageSelected: '10.jpeg',
    });

    const { getAllByTestId } = render(<FormCreditCard />);

    const imageCardElement = getAllByTestId('image-card-test')[0];

    fireEvent.click(imageCardElement);

    expect(imageCardElement).toBeTruthy();
  });

  it('should be able to simulate error when field number is empty', async () => {
    mockedError.mockImplementation(() => {
      throw new Error();
    });

    (useCreditCard as unknown as jest.Mock).mockReturnValue({
      handleSetValuesCreditCard: () => jest.fn(),
      handleTransitionCreditCard: () => jest.fn(),
      handleFocusField: () => jest.fn(),
      handleSaveFlagCard: () => jest.fn(),
      handleCreateCreditCard: () => jest.fn(),
      creditCard: mockCreditCard,
    });

    const { getByTestId, getByText } = render(<FormCreditCard />);

    const numberField = getByTestId('input-number');
    const buttonElement = getByText('Create new card');

    fireEvent.change(numberField, { target: { value: null } });

    fireEvent.click(buttonElement);

    await waitFor(() => {
      expect.objectContaining({
        type: 'error',
      });
    });
  });
});
