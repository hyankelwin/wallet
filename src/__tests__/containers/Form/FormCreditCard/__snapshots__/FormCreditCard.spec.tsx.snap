// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCreditCard container should be able to add new credit card on FormCreditCard container with success 1`] = `
<DocumentFragment>
  <form>
    <div
      class="sc-hKwDye bTTcgA"
    >
      <div
        class="sc-eCImPb gmgBGq"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb jocPJw"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb kYySTF"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb lmsHkS"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb cYTlPc"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb eJpoZl"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb bMRswX"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb eUyLNK"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb euxoHt"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb iuwLGw"
        data-testid="image-card-test"
      />
    </div>
    <div
      class="sc-bdvvtL cdqBYk"
    >
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Card number
        </span>
        <div
          class="sc-gKclnd heEQIa"
          data-testid="input-container"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 2l-2 2m-7.61 7.61a5.5 5.5 0 1 1-7.778 7.778 5.5 5.5 0 0 1 7.777-7.777zm0 0L15.5 7.5m0 0l3 3L22 7l-3-3m-3.5 3.5L19 4"
            />
          </svg>
          <input
            data-testid="input-number"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Card name
        </span>
        <div
          class="sc-gKclnd heEQIa"
          data-testid="input-container"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
          <input
            data-testid="input-name"
            style="text-transform: uppercase;"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Expiration date
        </span>
        <div
          class="row"
        >
          <div
            class="expiration-input"
          >
            <div
              class="sc-furwcr kYUlhF"
              data-testid="input-select-container"
            >
              <select
                data-testid="input-month"
              >
                <option
                  hidden=""
                  value=""
                >
                  Month
                </option>
                <option
                  value="01"
                >
                  01
                </option>
                <option
                  value="02"
                >
                  02
                </option>
                <option
                  value="03"
                >
                  03
                </option>
                <option
                  value="04"
                >
                  04
                </option>
                <option
                  value="05"
                >
                  05
                </option>
                <option
                  value="06"
                >
                  06
                </option>
                <option
                  value="07"
                >
                  07
                </option>
                <option
                  value="08"
                >
                  08
                </option>
                <option
                  value="09"
                >
                  09
                </option>
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="11"
                >
                  11
                </option>
                <option
                  value="12"
                >
                  12
                </option>
              </select>
            </div>
            <div
              class="sc-furwcr kYUlhF"
              data-testid="input-select-container"
            >
              <select
                data-testid="input-year"
              >
                <option
                  hidden=""
                  value=""
                >
                  Year
                </option>
                <option
                  value="2022"
                >
                  2022
                </option>
                <option
                  value="2023"
                >
                  2023
                </option>
                <option
                  value="2024"
                >
                  2024
                </option>
                <option
                  value="2025"
                >
                  2025
                </option>
                <option
                  value="2026"
                >
                  2026
                </option>
                <option
                  value="2027"
                >
                  2027
                </option>
                <option
                  value="2028"
                >
                  2028
                </option>
                <option
                  value="2029"
                >
                  2029
                </option>
                <option
                  value="2030"
                >
                  2030
                </option>
                <option
                  value="2031"
                >
                  2031
                </option>
                <option
                  value="2032"
                >
                  2032
                </option>
              </select>
            </div>
          </div>
          <div
            class="cvv"
          >
            <div
              class="sc-gKclnd heEQIa"
              data-testid="input-container"
            >
              <input
                data-testid="input-cvv"
                placeholder="CVV"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsDKAQ hrNEzd"
      >
        <button
          class="sc-jrQzAO kdZdkP"
          color="#FFFFFF"
          type="submit"
        >
          Create new card
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`FormCreditCard container should be able to add new credit card on FormCreditCard when year equal current 1`] = `
<DocumentFragment>
  <form>
    <div
      class="sc-hKwDye bTTcgA"
    >
      <div
        class="sc-eCImPb gmgBGq"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb jocPJw"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb kYySTF"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb lmsHkS"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb cYTlPc"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb eJpoZl"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb bMRswX"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb eUyLNK"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb euxoHt"
        data-testid="image-card-test"
      />
      <div
        class="sc-eCImPb iuwLGw"
        data-testid="image-card-test"
      />
    </div>
    <div
      class="sc-bdvvtL cdqBYk"
    >
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Card number
        </span>
        <div
          class="sc-gKclnd heEQIa"
          data-testid="input-container"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 2l-2 2m-7.61 7.61a5.5 5.5 0 1 1-7.778 7.778 5.5 5.5 0 0 1 7.777-7.777zm0 0L15.5 7.5m0 0l3 3L22 7l-3-3m-3.5 3.5L19 4"
            />
          </svg>
          <input
            data-testid="input-number"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Card name
        </span>
        <div
          class="sc-gKclnd heEQIa"
          data-testid="input-container"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
          <input
            data-testid="input-name"
            style="text-transform: uppercase;"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-dkPtRN hnnHcG"
      >
        <span>
          Expiration date
        </span>
        <div
          class="row"
        >
          <div
            class="expiration-input"
          >
            <div
              class="sc-furwcr kYUlhF"
              data-testid="input-select-container"
            >
              <select
                data-testid="input-month"
              >
                <option
                  hidden=""
                  value=""
                >
                  Month
                </option>
                <option
                  value="01"
                >
                  01
                </option>
                <option
                  value="02"
                >
                  02
                </option>
                <option
                  value="03"
                >
                  03
                </option>
                <option
                  value="04"
                >
                  04
                </option>
                <option
                  value="05"
                >
                  05
                </option>
                <option
                  value="06"
                >
                  06
                </option>
                <option
                  value="07"
                >
                  07
                </option>
                <option
                  value="08"
                >
                  08
                </option>
                <option
                  value="09"
                >
                  09
                </option>
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="11"
                >
                  11
                </option>
                <option
                  value="12"
                >
                  12
                </option>
              </select>
            </div>
            <div
              class="sc-furwcr kYUlhF"
              data-testid="input-select-container"
            >
              <select
                data-testid="input-year"
              >
                <option
                  hidden=""
                  value=""
                >
                  Year
                </option>
                <option
                  value="2022"
                >
                  2022
                </option>
                <option
                  value="2023"
                >
                  2023
                </option>
                <option
                  value="2024"
                >
                  2024
                </option>
                <option
                  value="2025"
                >
                  2025
                </option>
                <option
                  value="2026"
                >
                  2026
                </option>
                <option
                  value="2027"
                >
                  2027
                </option>
                <option
                  value="2028"
                >
                  2028
                </option>
                <option
                  value="2029"
                >
                  2029
                </option>
                <option
                  value="2030"
                >
                  2030
                </option>
                <option
                  value="2031"
                >
                  2031
                </option>
                <option
                  value="2032"
                >
                  2032
                </option>
              </select>
            </div>
          </div>
          <div
            class="cvv"
          >
            <div
              class="sc-gKclnd heEQIa"
              data-testid="input-container"
            >
              <input
                data-testid="input-cvv"
                placeholder="CVV"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsDKAQ hrNEzd"
      >
        <button
          class="sc-jrQzAO kdZdkP"
          color="#FFFFFF"
          type="submit"
        >
          Create new card
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
